src/rules/no-self-import.js

```diff
@@ -0,0 +1,42 @@
+/**
+ * @fileOverview Prohibits a file from importing itself
+ * @author Gio d'Amelio
+ */
+
+import resolve from 'eslint-module-utils/resolve'
+import isStaticRequire from '../core/staticRequire'
+
+function isImportingSelf(context, node, requireName) {
+  const filePath = context.getFilename()
+  const resolvedPath = resolve(requireName, context)
+  if (filePath === resolvedPath) {
+    context.report({
+        node,
+        // TODO: check in about message, should it print the name of the module itself
+        message: 'Module imports itself',
```

Comment:
> should it print the name of the module itself

I think there is no need. If they see the error in the editor, they will already have the file open and see which file the error is for. If they see it in the terminal or a CI environment's terminal, ESLint will display the location of the file it is in.